File: apps/ios/ScribbleFit/ScribbleFit/Core/Network/Models/ParsedWorkoutDto.swift
import Foundation

public struct ParsedWorkoutDto: Codable, Sendable {
    public let date: String
    public let location: String?
    public let exercises: [ParsedExerciseDto]
    
    public init(date: String, location: String? = nil, exercises: [ParsedExerciseDto]) {
        self.date = date
        self.location = location
        self.exercises = exercises
    }
}

public struct ParsedExerciseDto: Codable, Sendable {
    public let canonicalName: String
    public let sets: [ParsedSetDto]
    
    public init(canonicalName: String, sets: [ParsedSetDto]) {
        self.canonicalName = canonicalName
        self.sets = sets
    }
}

public struct ParsedSetDto: Codable, Sendable {
    public let weight: Double
    public let reps: Int
    public let rpe: Double?
    public let notes: String?
    
    public init(weight: Double, reps: Int, rpe: Double? = nil, notes: String? = nil) {
        self.weight = weight
        self.reps = reps
        self.rpe = rpe
        self.notes = notes
    }
}

File: apps/android/core/network/src/main/java/com/scribblefit/core/network/model/ParsedWorkoutDto.kt
package com.scribblefit.core.network.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ParsedWorkoutDto(
    val date: String,
    val location: String? = null,
    val exercises: List<ParsedExerciseDto>
)

@Serializable
data class ParsedExerciseDto(
    @SerialName("canonical_name")
    val canonicalName: String,
    val sets: List<ParsedSetDto>
)

@Serializable
data class ParsedSetDto(
    val weight: Double,
    val reps: Int,
    val rpe: Double? = null,
    val notes: String? = null
)
